lib/spack/spack/builder.py

```diff
@@ -244,6 +244,8 @@ def __new__(mcs, name, bases, attr_dict):
                 callbacks_from_base = getattr(base, temporary_stage.attribute_name, None)
                 if callbacks_from_base:
                     break
+            else:
+                callbacks_from_base = []
             callbacks_from_base = callbacks_from_base or []
```

Comment:
Are you sure `callbacks_from_base = callbacks_from_base or []` should stay?

Reply:
This is redundant now and can be removed, thanks